libtenzir/builtins/health-metrics/memory.cpp

```diff
@@ -0,0 +1,52 @@
+//    _   _____   __________
+//   | | / / _ | / __/_  __/     Visibility
+//   | |/ / __ |_\ \  / /          Across
+//   |___/_/ |_/___/ /_/       Space and Time
+//
+// SPDX-FileCopyrightText: (c) 2023 The Tenzir Contributors
+// SPDX-License-Identifier: BSD-3-Clause
+
+#include <tenzir/plugin.hpp>
+#include <tenzir/type.hpp>
+
+#include <caf/typed_event_based_actor.hpp>
+
+#include <filesystem>
+
+namespace tenzir::plugins::health_ram {
+
+namespace {
+
+auto get_raminfo() -> caf::expected<record> {
+  static auto pagesize = ::sysconf(_SC_PAGESIZE);
+  auto phys_pages = ::sysconf(_SC_PHYS_PAGES);
+  auto available_pages = ::sysconf(_SC_AVPHYS_PAGES);
+  auto result = record{};
+  result["total_bytes"] = phys_pages * pagesize;
+  result["available_bytes"] = available_pages * pagesize;
+  return result;
+}
+
+class plugin final : public virtual health_metrics_plugin {
+public:
+  auto name() const -> std::string override {
+    return "health-ram";
+  }
+
+  auto make_collector() const -> collector override {
+    return get_raminfo;
+  }
+
+  static auto metrics_layout() -> record_type /*override*/ {
+    return record_type{{
+      {"total_bytes", uint64_type{}},
+      {"available_bytes", uint64_type{}},
```

Comment:
Use dashes.

Reply:
The resulting events should surely use underscores instead of dashes. Thus, I think it's more consistent to go the other direction and use underscores everywhere.